### DISCUSSION TITLE
Jest upgrade is failing due to some dependencies version incompatibilities

### DISCUSSION THREAD

**Discussion by jomarko** [renovatebot/renovate#17426] [Request Help]
Hello, Renovate prepared 'jest' upgrade PR in our project [1], however that PR can not be compiled with the following error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: smartevents-ui@0.0.1
npm ERR! Found: jest@29.0.0
npm ERR! node_modules/jest
npm ERR!   dev jest@"29.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer jest@"^28.0.0" from ts-jest@28.0.8
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"28.0.8" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/runner/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-08-2[6](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/runs/8029239781?check_suite_focus=true#step:4:7)T03_22_02_[7](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/runs/8029239781?check_suite_focus=true#step:4:8)[9](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/runs/8029239781?check_suite_focus=true#step:4:10)6Z-debug-0.log
```

I would like to ask what is general approach in similar situation, when Renovate tries to upgrade some group of dependencies to latest possible, but it results to unsatisfied dependency issue.

[1] https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/144

**Comment by viceice** [comment]
use dashboard approval for major version and use our recommend group. then wait until all deps are compatible before approval

**Reply by jomarko** [reply]
@viceice let me check my understanding of your suggestion please.

So you suggest  we should group all major updates into single PR and regularly recheck if the current major versions are compatible?

**Reply by rarkins** [reply]
No, we already have a grouping rule for Jest-related dependencies which is part of `config:base`. You should only group major dependencies when they are related/need to be updated together, such as Jest and ts-jest.

Second part is dashboard approval. We recommend you use dependencyDashboardApproval=true for major updates from Renovate, like we do in this repo.

**Reply by jomarko** [reply]
@rarkins the renovate opened PR where both `jest` and `ts-jest` are upgraded. Could you help me to understand why the proposed upgrade by Renovate results in some incompatibilities please? Repeating our PR link for easier reference https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/144

**Reply by rarkins** [reply]
No, I can't help you with that. It's an npm or jest issue and not renovate. Quite possibly you just need to be patient for one missing dependency to be compatible with v29.

**Reply by jomarko** [reply]
@rarkins yes, I am not saying it is an issue of Renovate. as new user of renovate, just trying to find out what is the usual solution/approach to the similar situation. As you stated, I need probably just wait with upgrade of jest* dependencies.

**Reply by rarkins** [reply]
The npm message you pasted earlier  is as descriptive as you will likely get. It indicates the problem relates to ts-jest v28

**Reply by viceice** [reply]
you need to wait for ts-jest v29
